### ISSUE TITLE
flatMap with take(1) doesn't unsubscribe from source

### ISSUE THREAD

**Issue by tehsenaus** [pozadi/kefir#92]
Hi,

I've been hunting memory leaks in my app this week, and I think I've tracked one down to Kefir. It seems that flatMap will unsubscribe once take(1) is done, but then resubscribes again, even though it should now be inactive.

I'm using 1.3.1, not sure if this bug is in v2.

This code will reproduce the problem:
```js
var subCount = 0, unsubCount = 0;
var u = kefir.fromBinder(function (em) {
    subCount++;
    em.emit(1);

    return function () {
        unsubCount++;
    }
})

var s = kefir.fromBinder(function (em) {
    em.emit(1);
}).flatMap(function () {
    return u;
}).take(1);

var data = [], ended = false;
s.onAny(function (ev) {
    if (ev.type === 'value') data.push(ev.value);
    else if ( ev.type === 'end') ended = true;
});

data.should.eql([1]);
ended.should.be.ok;

// subCount = 2, but unsubCount = 1
unsubCount.should.equal(subCount);
```

**Comment by pozadi** [comment]
Thanks! I know this ones are pretty hard to hunt (was fixing couple of similar already), and reproducing code was very helpful.

Is it critical to you to have a fix in v1.x.x?

**Comment by pozadi** [comment]
Fixed in `2.0.1` and `1.3.2`

**Comment by tehsenaus** [comment]
Thanks for the quick fix!
